src/renderers/dom/ReactDOMStringRenderer.js

```diff
@@ -36,10 +40,13 @@ function renderToString(element) {
  * See https://facebook.github.io/react/docs/react-dom-server.html#rendertostaticmarkup
  */
 function renderToStaticMarkup(element) {
-  invariant(
-    React.isValidElement(element),
-    'renderToStaticMarkup(): You must pass a valid ReactElement.',
-  );
+  const disableNewFiberFeatures = ReactFeatureFlags.disableNewFiberFeatures;
+  if (disableNewFiberFeatures) {
+    invariant(
```

Comment:
React.children.toArray will accepts valid element or array.
1.  When children is single element: It will be validated by `React.isValidElement`. otherwise will throw error.
2.  When children is array type: It will be validated whether the array is valid, It can either array(native array) or object that have iterator function. and per element of array will be validated by React.isValidElement. otherwise will throw error.


So we can validated the following cases by `React.children.toArray`
1.  First call to render, `ReactDOMServer.renderToString(reactElement)` 
2.  Element be returned by render method.